### PR TITLE
[ie/Radio1Be] Add extractor

### LINKED ISSUE #8707: Site request radio1.be
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

World

### Example URLs

https://radio1.be/luister/select/de-ochtend/komt-n-va-volgend-jaar-op-in-wallonie

### Provide a description that is worded well enough to be understood

Site support request for Radio 1, a Belgiann radio channel operated by the Flemish public broadcaster Vlaamse Radio- en Televisieomroep (VRT)

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://radio1.be/luister/select/de-ochtend/komt-n-va-volgend-jaar-op-in-wallonie']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2023.11.29.232714 from yt-dlp/yt-dlp-nightly-builds [6a9c7a2b5] (linux_exe)
[debug] Python 3.10.13 (CPython x86_64 64bit) - Linux-6.2.0-37-generic-x86_64-with-glibc2.35 (OpenSSL 3.2.0 23 Nov 2023, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2
[debug] Optional libraries: Cryptodome-3.19.0, brotli-1.1.0, certifi-2023.11.17, mutagen-1.47.0, requests-2.31.0, secretstorage-3.3.3, sqlite3-3.44.2, urllib3-2.1.0, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1792 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2023.11.29.232714 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2023.11.29.232714 from yt-dlp/yt-dlp-nightly-builds)
[generic] Extracting URL: https://radio1.be/luister/select/de-ochtend/komt-n-va-volgend-jaar-op-in-wallonie
[generic] komt-n-va-volgend-jaar-op-in-wallonie: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] komt-n-va-volgend-jaar-op-in-wallonie: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://radio1.be/luister/select/de-ochtend/komt-n-va-volgend-jaar-op-in-wallonie
Traceback (most recent call last):
  File "yt_dlp/YoutubeDL.py", line 1570, in wrapper
  File "yt_dlp/YoutubeDL.py", line 1705, in __extract_info
  File "yt_dlp/extractor/common.py", line 717, in extract
  File "yt_dlp/extractor/generic.py", line 2531, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://radio1.be/luister/select/de-ochtend/komt-n-va-volgend-jaar-op-in-wallonie
```

### PR DESCRIPTION
**Pull request by HobbyistDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This PR will add support for `radio1.be` site. This site is subclassed from `VRTBaseIE`. The test may be fail because closed connection from remote or something like that (which i don't know is it only occur in my machine or not)

Fixes #8707

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by seproDev** [comment]
Could we also add support for articles that contain a video? https://radio1.be/lees/europese-unie-wil-onmiddellijke-humanitaire-pauze-en-duurzaam-staakt-het-vuren-in-gaza?view=web

File: yt_dlp/extractor/vrt.py [hunk 1]
```diff
@@ -425,3 +427,74 @@ def _real_extract(self, url):
...
+            'thumbnail': r're:https?://cds\.vrt\.radio/[^/#\?&]+',
+            'display_id': 'europese-unie-wil-onmiddellijke-humanitaire-pauze-en-duurzaam-staakt-het-vuren-in-gaza'
+        },
+        'playlist_mincount': 1
+    }]
+
+    def _extract_video_entries(self, next_js_data, display_id):
+        entries = []
+        video_data = variadic(traverse_obj(
+            next_js_data, {lambda x: x if x.get('mediaReference') else traverse_obj(x, ('paragraphs', ...))}))
```
Comment (seproDev) [inline yt_dlp/extractor/vrt.py:459]: We can already check that all entries have a mediaReference
```suggestion
        video_data = traverse_obj(
            next_js_data, ((None, ('paragraphs', ...)), {lambda x: x if x['mediaReference'] else None}))
```

File: yt_dlp/extractor/vrt.py [hunk 2]
```diff
@@ -425,3 +427,74 @@ def _real_extract(self, url):
...
+        for data in video_data:
+            media_reference = data.get('mediaReference')
+            if media_reference is None:
+                continue
+            formats, subtitles = self._extract_formats_and_subtitles(self._call_api(media_reference), display_id)
```
Comment (seproDev) [inline yt_dlp/extractor/vrt.py:464]:
```suggestion
            media_reference = data['mediaReference']
            formats, subtitles = self._extract_formats_and_subtitles(
                self._call_api(media_reference), display_id)
```

File: yt_dlp/extractor/vrt.py [hunk 3]
```diff
@@ -425,3 +427,74 @@ def _real_extract(self, url):
...
+
+            entries.append({
+                'id': media_reference,
+                'formats': formats,
+                'subtitles': subtitles,
+                **traverse_obj(data, {
+                    'title': 'title',
```
Comment (seproDev) [inline yt_dlp/extractor/vrt.py:471]: Just to be sure, let's also validate the type
```suggestion
                    'title': ('title', {str}),
```

File: yt_dlp/extractor/vrt.py [hunk 4]
```diff
@@ -425,3 +427,74 @@ def _real_extract(self, url):
...
+                'description': self._html_search_meta(['description', 'og:description', 'twitter:description'], webpage),
+                'thumbnail': self._html_search_meta(['og:image', 'twitter:image'], webpage)
+            }
+        )
+        return self.playlist_result(
+            entries, playlist_id=playlist_info['id'], display_id=display_id, **traverse_obj(playlist_info, {
+                'thumbnail': 'thumbnail',
+                'playlist_description': 'description',
+                'playlist_title': 'title'
+            }))
```
Comment (seproDev) [inline yt_dlp/extractor/vrt.py:500]:
```suggestion
        return self.playlist_result(
            entries, **merge_dicts(traverse_obj(
                next_js_data, ({
                    'id': ('id', {str}),
                    'title': ('title', {str}),
                    'description': (('description', 'content'), {clean_html}),
                }), get_all=False), {
                    'display_id': display_id,
                    'title': self._html_search_meta(['name', 'og:title', 'twitter:title'], webpage),
                    'description': self._html_search_meta(['description', 'og:description', 'twitter:description'], webpage),
                    'thumbnail': self._html_search_meta(['og:image', 'twitter:image'], webpage)}))
```

File: yt_dlp/extractor/vrt.py [hunk 5]
```diff
@@ -425,3 +427,74 @@ def _real_extract(self, url):
                 ['description', 'twitter:description', 'og:description'], webpage),
             '_old_archive_ids': [make_archive_id('Canvas', video_id)],
         }
+
+
+class Radio1BeIE(VRTBaseIE):
+    _VALID_URL = r'https?://radio1\.be/\w+(/[\w-]+/[\w-]+)?/(?P<display_id>[\w-]+)'
```
Comment (seproDev) [inline yt_dlp/extractor/vrt.py:433]: How about, like this?
```suggestion
    _VALID_URL = r'https?://radio1\.be/(?:lees|luister/select)/(?P<display_id>[\w/-]+)'
```

File: yt_dlp/extractor/vrt.py [hunk 6]
```diff
@@ -425,3 +426,66 @@ def _real_extract(self, url):
...
+            entries.append({
+                'id': media_reference,
+                'formats': formats,
+                'subtitles': subtitles,
+                **traverse_obj(data, {
+                    'title': ('title', {str}),
+                    'description': ('body', {clean_html})
+                })
+            })
+        return entries
```
Comment (bashonly) [inline yt_dlp/extractor/vrt.py:473]:
```suggestion
    def _extract_video_entries(self, next_js_data, display_id):
        video_data = traverse_obj(
            next_js_data, ((None, ('paragraphs', ...)), {lambda x: x if x['mediaReference'] else None}))
        for data in video_data:
            media_reference = data['mediaReference']
            formats, subtitles = self._extract_formats_and_subtitles(
                self._call_api(media_reference), display_id)

            yield {
                'id': media_reference,
                'formats': formats,
                'subtitles': subtitles,
                **traverse_obj(data, {
                    'title': ('title', {str}),
                    'description': ('body', {clean_html})
                }),
            }
```

File: yt_dlp/extractor/vrt.py [hunk 7]
```diff
@@ -425,3 +426,66 @@ def _real_extract(self, url):
...
+
+    def _real_extract(self, url):
+        display_id = self._match_valid_url(url).group('display_id')
+        webpage = self._download_webpage(url, display_id)
+        next_js_data = self._search_nextjs_data(webpage, display_id)['props']['pageProps']['item']
+
+        entries = self._extract_video_entries(next_js_data, display_id)
+        return self.playlist_result(
+            entries, **merge_dicts(traverse_obj(
```
Comment (bashonly) [inline yt_dlp/extractor/vrt.py:482]:
```suggestion
        return self.playlist_result(
            self._extract_video_entries(next_js_data, display_id), **merge_dicts(traverse_obj(
```

File: yt_dlp/extractor/vrt.py [hunk 8]
```diff
@@ -425,3 +426,66 @@ def _real_extract(self, url):
...
+                next_js_data, ({
+                    'id': ('id', {str}),
+                    'title': ('title', {str}),
+                    'description': (('description', 'content'), {clean_html}),
+                }), get_all=False), {
+                    'display_id': display_id,
+                    'title': self._html_search_meta(['name', 'og:title', 'twitter:title'], webpage),
+                    'description': self._html_search_meta(['description', 'og:description', 'twitter:description'], webpage),
+                    'thumbnail': self._html_search_meta(['og:image', 'twitter:image'], webpage)}))
```
Comment (bashonly) [inline yt_dlp/extractor/vrt.py:491]:
```suggestion
                    'thumbnail': self._html_search_meta(['og:image', 'twitter:image'], webpage),
                }))
```

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/_extractors.py [hunk 9]
```diff
@@ -2289,6 +2289,7 @@
     VrtNUIE,
     KetnetIE,
     DagelijkseKostIE,
+    Radio1BeIE
```
Comment (bashonly) [inline yt_dlp/extractor/_extractors.py:2292]:
```suggestion
    Radio1BeIE,
```

File: yt_dlp/extractor/vrt.py [hunk 10]
```diff
@@ -425,3 +426,64 @@ def _real_extract(self, url):
...
+                next_js_data, ({
+                    'id': ('id', {str}),
+                    'title': ('title', {str}),
+                    'description': (('description', 'content'), {clean_html}),
+                }), get_all=False), {
+                    'display_id': display_id,
+                    'title': self._html_search_meta(['name', 'og:title', 'twitter:title'], webpage),
+                    'description': self._html_search_meta(['description', 'og:description', 'twitter:description'], webpage),
+                    'thumbnail': self._html_search_meta(['og:image', 'twitter:image'], webpage),
+                }))
```
Comment (bashonly) [inline yt_dlp/extractor/vrt.py:489]:
```suggestion
            }))
```

**Pull request merged**